### PR TITLE
Feature: automate publishing

### PR DESCRIPTION
**Pull request by filoxo**

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -9,17 +9,30 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions/setup-node@v2
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
Comment (filoxo) [inline .github/workflows/build-test.yml:15]: necessary to pull all git tags.

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -9,17 +9,30 @@ on:
...
+          pnpm run lint
+          pnpm run check-format
+      - name: Build
+        run: pnpm run build
+      - name: Publish to npm
+        if: "github.ref == 'refs/heads/main' && !contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+        run: pnpm exec auto shipit --dry-run # TODO: remove dry-run once its mostly working
```
Comment (filoxo) [inline .github/workflows/build-test.yml:38]: ⚠️ note the `--dry-run` flag. This command shouldn't publish anything until that is removed.

**Comment by filoxo** [comment]
I was able to also run & test GitHub action locally using [act](https://github.com/nektos/act). I don't know that that's a workflow we want to support (would possibly allow for making canary releases a lot easier, but with the added cost of maintaining those versions). It just requires a .env file with a valid `GH_TOKEN` and `NPM_TOKEN` to be sourced before running act.

File: .github/workflows/build-test.yml [hunk 3]
```diff
@@ -9,17 +9,30 @@ on:
...
+          version: 7.26.0
+      - uses: actions/setup-node@v3
         with:
-          version: 5.14.1
-      - run: pnpm install --frozen-lockfile
-      - run: pnpm test
-      - run: pnpm run lint
-      - run: pnpm run check-format
-      - run: pnpm run build
+          node-version: 16
```
Comment (joeldenning) [inline .github/workflows/build-test.yml:21]: Maybe upgrade to node 18 or 19?

File: .github/workflows/build-test.yml [hunk 4]
```diff
@@ -9,17 +9,30 @@ on:
...
+      - name: Test
+        run: pnpm test
+      - name: Validate
+        run: |
+          pnpm run lint
+          pnpm run check-format
+      - name: Build
+        run: pnpm run build
+      - name: Publish to npm
+        if: "github.ref == 'refs/heads/main' && !contains(github.event.head_commit.message, 'ci skip') && !contains(github.event.head_commit.message, 'skip ci')"
```
Comment (joeldenning) [inline .github/workflows/build-test.yml:34]: Maybe also check here if the repo org name is `single-spa`, so that forks don't try to publish?

**Review by joeldenning — APPROVED** [review]
Looks good, a couple small, optional comments

**Comment by joeldenning** [comment]
How does `auto shipit` determine what new version to publish the changes under?

File: .github/workflows/build-test.yml (hunk 4 above)
Comment (filoxo) [reply .github/workflows/build-test.yml:34]: Good call!

**Comment by filoxo** [comment]
If I recall,`shipit` is a wrapper for multiple other commands, [`version`](https://intuit.github.io/auto/docs/generated/version) included. That makes use of the label applied to the PR per the docs here https://intuit.github.io/auto/docs/generated/version so it still requires manually specifying, but via a label on the PR instead of through publishing commands.

**Pull request merged**